### ISSUE TITLE
[consistent-type-imports] conflicts with no-duplicate-imports

### ISSUE THREAD

**Issue by yenshih** [typescript-eslint/typescript-eslint#2538]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "no-duplicate-imports": "error",
    "@typescript-eslint/consistent-type-imports": "error"
  }
}
```

```TS
import axios, {AxiosResponse} from 'axios';
// Import "AxiosResponse" are only used as types
```
```TS
import axios from 'axios';
import type {AxiosResponse} from 'axios';
// 'axios' import is duplicated
```

**Expected Result**

`@typescript-eslint/consistent-type-imports` is compatible with `no-duplicate-imports`

**Actual Result**

`@typescript-eslint/consistent-type-imports` conflicts with `no-duplicate-imports`

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.1.0` |
| `@typescript-eslint/parser`        | `4.1.0` |
| `TypeScript`                       | `4.0.2` |
| `ESLint`                           | `7.8.1` |
| `node`                             | `10.16.3` |

**Comment by bradzacher** [comment]
See https://github.com/typescript-eslint/typescript-eslint/issues/2315
